### DISCUSSION TITLE
FVM is reading from `fvc_config.json` but from 3.0.0 it should read from `.fvmrc` file.

### DISCUSSION THREAD

**Discussion by Rohithgilla12** [codemagic-ci-cd/codemagic-docs#2735] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

- Create flutter project
- Add fvm config using v3
- Chose from fvm config by following this https://docs.codemagic.io/flutter-configuration/flutter-projects/#setting-the-flutter-version for workflow builder.
- Run the build

![image](https://github.com/user-attachments/assets/93e07935-6451-4e5c-93b0-999b010dcf11)

### Expected results

The build to detect the flutter version from `.fvmrc` file and build accordingly.

### Actual results

The build fails because it couldn't find the `.fvm` folder and `fvm_config.json` file.
But from `fvm` v3  https://github.com/leoafarias/fvm/blob/main/CHANGELOG.md#300 they use  a custom `.fvmrc` file and it doesn't need having `.fvm` directory checked into git.

**Comment by dtrdic** [comment]
hey @Rohithgilla12 Codemagic can read the flutter version from both files `.fvm/fvm_cnofig.json` and `.fvmrc`.
Have you run` fvm use {version}` locally? The fvm config files will be automatically created when you run` fvm use {particular_version}`
 I don't know what exact error you have since you haven't provided the build id, but the payload of the json also changed from:
```
{
  "flutterSdkVersion": "stable"
}
```
to
```
{
  "flutter": "stable"
}
```
can you doublecheck if it's correct?

**Reply by Rohithgilla12** [reply]
Thanks, sorry missed the notification.

**Reply by Rohithgilla12** [reply]
```json
{
  "flutter": "3.24.2",
  "runPubGetOnSdkChanges": true,
  "updateVscodeSettings": true,
  "updateGitIgnore": true
}
```

Is my `.fvmrc` file
This is the build id `67002635b11e9de445767d11`

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.

**Reply by Rohithgilla12** [reply]
https://github.com/orgs/codemagic-ci-cd/discussions/2735#discussioncomment-10847195
